packages/valory/contracts/gnosis_safe/contract.py

```diff
@@ -449,15 +451,51 @@ def get_raw_safe_transaction(  # pylint: disable=too-many-arguments,too-many-loc
             and max_priority_fee_per_gas is None
         ):
             tx_parameters.update(ledger_api.try_get_gas_pricing(old_price=old_price))
-        # note, the next line makes an eth_estimateGas call!
         transaction_dict = w3_tx.buildTransaction(tx_parameters)
-        transaction_dict["gas"] = Wei(
-            max(transaction_dict["gas"] + 75000, configured_gas)
+        estimated_gas = cls._estimate_gas(
+            ledger_api,
+            transaction_dict,
         )
+        transaction_dict["gas"] = Wei(max(estimated_gas + 75000, configured_gas))
         transaction_dict["nonce"] = nonce  # pragma: nocover
 
         return transaction_dict
 
+    @staticmethod
+    def _estimate_gas(ledger_api: EthereumApi, tx: Dict) -> int:
+        """Estimate the gas for the given tx."""
```

Comment:
Would it make sense to move this method to the ledger instead?

Reply:
Yes, https://github.com/valory-xyz/open-aea/pull/268